### PR TITLE
Add fingerprints for --config auto

### PR DESCRIPTION
**Pull request by minusworld**
Test plan:
- ran fingerprints.yaml on semgrep-rules with severity set to INFO. (INVENTORY will not display results)

```
➜  semgrep-rules git:(fingerprinting-for-auto) ✗ semgrep --json -f fingerprints/fingerprints.yaml . | jq '.results[].check_id' | sort | uniq
Running 56 rules...
100%|█████|56/56
ran 56 rules on 2367 files: 1123 findings

A new version of Semgrep is available. See https://semgrep.dev/docs/upgrading
"fingerprints.cai-c-technology-c"
"fingerprints.cai-csharp-technology-csharp"
"fingerprints.cai-generic-technology-github-actions"
"fingerprints.cai-generic-technology-kubernetes"
"fingerprints.cai-generic-technology-nginx"
"fingerprints.cai-go-technology-go"
"fingerprints.cai-html-technology-html"
"fingerprints.cai-java-framework-spring"
"fingerprints.cai-java-technology-java"
"fingerprints.cai-javascript-framework-expressjs"
"fingerprints.cai-javascript-technology-javascript"
"fingerprints.cai-kotlin-technology-kotlin"
"fingerprints.cai-ocaml-technology-ocaml"
"fingerprints.cai-php-technology-php"
"fingerprints.cai-python-framework-django"
"fingerprints.cai-python-framework-flask"
"fingerprints.cai-python-technology-python"
"fingerprints.cai-ruby-technology-ruby"
"fingerprints.cai-scala-technology-scala"
"fingerprints.cai-shell-technology-bash"
"fingerprints.cai-shell-technology-sh"
"fingerprints.cai-terraform-technology-terraform"
"fingerprints.cai-typescript-technology-typescript"
```

### CODE REVIEW

File: fingerprints/fingerprints.yaml [hunk 1]
```diff
@@ -0,0 +1,821 @@
...
+    severity: INVENTORY
+  - id: cai-generic-technology-nginx
+    languages:
+      - generic
+    message: Detected a .conf file. This likely indicates that Nginx is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.conf"
```
Comment (clintgibler) [inline fingerprints/fingerprints.yaml:630]: Hm actually aren't there a variety of files that end in `.conf`? Could we make this more precise?

File: fingerprints/fingerprints.yaml [hunk 2]
```diff
@@ -0,0 +1,821 @@
...
+    severity: INVENTORY
+  - id: cai-shell-technology-bash
+    languages:
+      - generic
+    message: Detected a bash script. This likely indicates bash is being used.
+    metadata:
+      license: Semgrep Registry License
+    pattern-either:
+      - pattern: "#!/bin/bash"
+      - pattern: "#!/usr/bin/env ... bash"
```
Comment (clintgibler) [inline fingerprints/fingerprints.yaml:652]: Should this look for file extension? Also, could the pattern use the new Bash mode?

File: fingerprints/fingerprints.yaml [hunk 3]
```diff
@@ -0,0 +1,821 @@
...
+    severity: INVENTORY
+  - id: cai-php-technology-php
+    languages:
+      - generic
+    message: Detected a .php file. This likely indicates PHP is in use.
+    metadata:
+      license: Semgrep Registry License
+    paths:
+      include:
+        - "*.php"
```
Comment (clintgibler) [inline fingerprints/fingerprints.yaml:762]: I think PHP files also end in php4 or php5 or something? Looked it up: https://www.guru99.com/what-is-php-first-php-program.html

.php3
.php4
.php5
.phps

Could also look for `<?php ...>`

**Review by clintgibler — APPROVED** [review]
Maybe some improvements that can be made but overall lgtm.

File: fingerprints/fingerprints.yaml (hunk 2 above)
Comment (minusworld) [reply fingerprints/fingerprints.yaml:652]: I don't think the bash parser would pick up comments

File: fingerprints/fingerprints.yaml (hunk 1 above)
Comment (minusworld) [reply fingerprints/fingerprints.yaml:630]: It came this way; I think it's probably OK for now

**Pull request merged**